src/app/api/backers/[backer]/tx-history/context/route.ts

```diff
@@ -12,6 +18,63 @@ function parseMultiParam(search: URLSearchParams, key: string): string[] {
     .filter(Boolean)
 }
 
+function parsePriceParams(search: URLSearchParams): PriceParam[] {
+  const rawPrices = search.getAll('price')
+  const prices: PriceParam[] = []
+
+  for (const raw of rawPrices) {
+    const [token, priceStr, decimalsStr] = raw.split(':')
+    if (!token || !priceStr) continue
+
+    const priceUsd = Number(priceStr)
+    const decimals = decimalsStr ? Number(decimalsStr) : 18
+
+    if (!Number.isFinite(priceUsd) || !Number.isFinite(decimals)) continue
+
+    prices.push({
+      token: token.toLowerCase(),
+      priceUsd,
+      decimals,
+    })
+  }
+
+  return prices
+}
+
+function buildClaimAmountUsdExpression(prices: PriceParam[]) {
+  if (!prices.length) {
+    return db.raw('0::numeric as "amountUsd"')
+  }
+
+  let sql = 'amount::numeric * (CASE convert_from("rewardToken", \'utf8\')'
+  const bindings: Array<string | number> = []
+
+  for (const { token, priceUsd, decimals } of prices) {
+    const factor = priceUsd / Math.pow(10, decimals || 18)
```

Comment:
```suggestion
    const factor = priceUsd / Math.pow(10, decimals)
```
Its not an optional value and the validation was done before.